src/DocBlock/Tags/Factory/MethodParameterFactory.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Reflection\DocBlock\Tags\Factory;
+
+use phpDocumentor\Reflection\DocBlock\Tags\Formatter;
+use function str_repeat;
+use function strlen;
+
+class MethodParameterFactory
+{
+    /**
+     * Formats the given default value to a string-able mixin
+     */
+    public function format($defaultValue): string
+    {
+        if (method_exists($this, $method = 'format'.ucfirst(gettype($defaultValue)))) {
+            return ' = ' . $this->{$method}($defaultValue);
+        }
+        return '';
+    }
+
+    protected function formatDouble(float $defaultValue): string
```

Comment:
I do prefer `private` for all new methods as we do not want others to interfere with this code. This make it easier to refactor methods as private methods are internal only, and we can remove them at will.

Reply:
All good